docs/source/blogs/tech_blog/blog12_Inference_Time_Compute_Implementation_in_TensorRT-LLM.md

```diff
@@ -0,0 +1,425 @@
+
+        yield tasks
+```
+Essentially, `process()` is an iterator in python that can return a list of tasks using yield statement. When the iterator is re-entered, that is, when the yield statement ends, the `Tasks` have been completed. That means the result of the `Task` has been written into its result field. Then the `process()` can proceed to the next steps.
+
+
+From here we can see that the implement of the `Controller` can focus on the design of the workflow. It does not directly call the `Worker` and does not need to care about how these tasks are completed. And that is how `Scaffolding` decouple inference-time compute method and execution backend.
+
+
+Also, `Controller` makes the inference-time compute method modular and reusable. It only requires the `sub-Controller` to be a member of class, and then the `process()` function of the `sub-Controller` is called using the “yield from” statement.
+```python
+yield from self.reward_controller.process(generation_tasks,
+                                                **reward_kwargs)
+```
+
+
+For the concurrency with ease of use, `Controller` provides two ways. As the code above shows, the yield statement yield a list of `Task`. So the first one is that the multiple Tasks in a yield statement is executed in parallel. The second way is for the multiple `sub-Controller` which can be executed in parallel. `Controller` provides syntactic sugar called `ParallelProcess`.
+```python
+generation_controllers = [
+            self.generation_controller for _ in range(sample_num)
+        ]
+        generation_kwargs_list = [generation_kwargs for _ in range(sample_num)]
+        generation_tasks = [copy.deepcopy(task) for _ in range(sample_num)]
+
+        yield ParallelProcess(generation_controllers,
+                              [[t] for t in generation_tasks],
+                              generation_kwargs_list)
+```
+
+
+#### ScaffoldingLlm
+With `Controller` and `Worker`, we still need something that can combine them together, that is the `ScaffoldingLlm` class.
+```python
+llm_worker = TRTLLMWorker.init_with_new_llm(
+    args.model_dir,
+    backend="pytorch",
+    max_batch_size=32,
+    max_num_tokens=4096,
+)
+
+prototype_controller = NativeGenerationController(sampling_params={
+    "temperature": 0.9,
+    "max_tokens": 1024,
+})
+
+llm = ScaffoldingLlm(
+    prototype_controller,
+    {NativeGenerationController.WorkerTag.GENERATION: proposer_worker},
+)
+results = llm.generate(prompts)
+```
+Users need to first create instances of `Worker` and `Controller`, and map them by `WorkerTag` to create the `ScaffoldingLlm` class. Then call the generate interface of `ScaffoldingLlm` to get the final result. 
+
+
+`ScaffoldingLlm` also provides async interface.
+```python
+async for result in llm.generate_async(prompt):
+    print(">>>", result.outputs[0].text)
+```
+So an instance of `ScaffoldingLlm` can support the concurrent execution of multiple requests.
```

Comment:
suggestion: Therefore, an instance of `ScaffoldingLlm` supports concurrent execution of multiple requests.